contracts/helpers/ChainlinkCalculator.sol

```diff
@@ -15,33 +15,38 @@ contract ChainlinkCalculator {
     /// e.g. for DAI-ETH oracle, inverse=false means that we request DAI price in ETH
     /// and inverse=true means that we request ETH price in DAI
-    /// @return Result Token price times amount
-    function singlePrice(AggregatorInterface oracle, uint256 inverseAndSpread, uint256 amount) external view returns(uint256) {
+    /// @return Amount * spread * oracle price
+    function singlePrice(AggregatorV3Interface oracle, uint256 inverseAndSpread, uint256 amount) external view returns(uint256) {
+        (, int256 latestAnswer,, uint256 latestTimestamp,) = oracle.latestRoundData();
         // solhint-disable-next-line not-rely-on-time
-        require(oracle.latestTimestamp() + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data");
+        require(latestTimestamp + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data");
         bool inverse = inverseAndSpread & _INVERSE_MASK > 0;
         uint256 spread = inverseAndSpread & (~_INVERSE_MASK);
         if (inverse) {
-            return amount * spread * 1e18 / oracle.latestAnswer().toUint256() / _SPREAD_DENOMINATOR;
+            return amount * spread * (10 ** oracle.decimals()) / latestAnswer.toUint256() / _SPREAD_DENOMINATOR;
         } else {
-            return amount * spread * oracle.latestAnswer().toUint256() / 1e18 / _SPREAD_DENOMINATOR;
+            return amount * spread * latestAnswer.toUint256() / (10 ** oracle.decimals()) / _SPREAD_DENOMINATOR;
         }
     }
 
     /// @notice Calculates price of token A relative to token B. Note that order is important
     /// @return Result Token A relative price times amount
-    function doublePrice(AggregatorInterface oracle1, AggregatorInterface oracle2, uint256 spread, uint256 amount) external view returns(uint256) {
+    function doublePrice(AggregatorV3Interface oracle1, AggregatorV3Interface oracle2, uint256 spread, uint256 amount) external view returns(uint256) {
+        require(oracle1.decimals() == oracle2.decimals(), "CC: oracle decimals don't match");
+
+        (, int256 latestAnswer1,, uint256 latestTimestamp1,) = oracle1.latestRoundData();
+        (, int256 latestAnswer2,, uint256 latestTimestamp2,) = oracle2.latestRoundData();
         // solhint-disable-next-line not-rely-on-time
-        require(oracle1.latestTimestamp() + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data O1");
+        require(latestTimestamp1 + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data O1");
         // solhint-disable-next-line not-rely-on-time
-        require(oracle2.latestTimestamp() + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data O2");
+        require(latestTimestamp2 + _ORACLE_EXPIRATION_TIME > block.timestamp, "CC: stale data O2");
 
-        return amount * spread * oracle1.latestAnswer().toUint256() / oracle2.latestAnswer().toUint256() / _SPREAD_DENOMINATOR;
+        return amount * spread * latestAnswer1.toUint256() / latestAnswer2.toUint256() / _SPREAD_DENOMINATOR;
```

Comment:
This line expects oracle decimals to match. Is is true for all the Chainlink oracles?

Reply:
We have line 41 ensuring decimals match